gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/BucketRegion.java

```diff
@@ -2723,12 +2723,10 @@ else if (GemFireCacheImpl.gfxdSystem()) {
       this.partitionedRegion.getPrStats().incDataStoreEntryCount(-sizeBeforeClear);
       prDs.updateMemoryStats(-oldMemValue);
     }
-    // explicitly clear overflow counters if no diskRegion is present
+    // explicitly clear overflow counters 
     // (for latter the counters are cleared by DiskRegion.statsClear)
-    if (getDiskRegion() == null) {
-      this.numOverflowOnDisk.set(0);
-      this.numOverflowBytesOnDisk.set(0);
-    }
+    this.numOverflowOnDisk.set(0);
```

Comment:
Not sure about moving out numOverflowOnDisk too. See DiskRegion.clearStats which is called on truncate. It reduces current size (and does not zero it), so if it gets invoked after this method then it will make this negative. I had added the getDiskRegion() check for that reason but failed to notice that numOverflowBytesOnDisk is not being reduced by statsClear. Will suggest moving out just numOverflowBytesOnDisk outside of the (getDiskRegion() == null) check and keeping it for numOverflowOnDisk

Reply:
Perhaps better option is to leave this code untouched, rather add call to "incNumOverflowBytesOnDisk()" in DiskRegion.statsClear with appropriate negative argument as being done for others there. This will make things uniform (like clearing this from the DiskRegion.statsClose too).

Reply:
As discussed, made the appropriate changes.